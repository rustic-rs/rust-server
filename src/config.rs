pub mod auth_config;
pub mod auth_file;
pub mod server_config;
