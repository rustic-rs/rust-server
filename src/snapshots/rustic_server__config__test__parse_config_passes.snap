---
source: src/config.rs
expression: config
---
RusticServerConfig {
    server: ConnectionSettings {
        listen: "127.0.0.1:8000",
    },
    storage: StorageSettings {
        data_dir: Some(
            "./test_data/test_repos/",
        ),
        quota: Some(
            0,
        ),
    },
    auth: HtpasswdSettings {
        disable_auth: false,
        htpasswd_file: Some(
            "/test_data/test_repo/.htpasswd",
        ),
    },
    acl: AclSettings {
        disable_acl: true,
        append_only: false,
        acl_path: Some(
            "/test_data/test_repo/acl.toml",
        ),
    },
    tls: TlsSettings {
        disable_tls: false,
        tls_key: Some(
            "/test_data/test_repo/key.pem",
        ),
        tls_cert: Some(
            "/test_data/test_repo/cert.pem",
        ),
    },
    log: LogSettings {
        log_level: Some(
            "info",
        ),
        log_file: Some(
            "/test_data/test_repo/rustic.log",
        ),
    },
}
